File: Lesson/Lesson 24 Four-way answerer experiment/Python/Button.py
import RPi.GPIO as GPIO
import time

BtnPin_1 = 22
LEDpin_1 = 18
BtnPin_2 = 23
LEDpin_2 = 19
BtnPin_3 = 24
LEDpin_3 = 20
BtnPin_4 = 25
LEDpin_4 = 21

GPIO.setmode(GPIO.BCM)
GPIO.setup(LEDpin_1,GPIO.OUT)
GPIO.setup(BtnPin_1,GPIO.IN)
GPIO.setup(LEDpin_2,GPIO.OUT)
GPIO.setup(BtnPin_2,GPIO.IN)
GPIO.setup(LEDpin_3,GPIO.OUT)
GPIO.setup(BtnPin_3,GPIO.IN)
GPIO.setup(LEDpin_4,GPIO.OUT)
GPIO.setup(BtnPin_4,GPIO.IN)
GPIO.output(LEDpin_1,GPIO.LOW)
GPIO.output(LEDpin_2,GPIO.LOW)
GPIO.output(LEDpin_3,GPIO.LOW)
GPIO.output(LEDpin_4,GPIO.LOW)

while True:
    if GPIO.input(BtnPin_1):
        time.sleep(0.04)
        while GPIO.input(BtnPin_1):
            GPIO.output(LEDpin_1,GPIO.HIGH)
            GPIO.output(LEDpin_2,GPIO.LOW)
            GPIO.output(LEDpin_3,GPIO.LOW)
            GPIO.output(LEDpin_4,GPIO.LOW)
    if GPIO.input(BtnPin_2):
        time.sleep(0.04)
        while GPIO.input(BtnPin_2):
            GPIO.output(LEDpin_2,GPIO.HIGH)
            GPIO.output(LEDpin_1,GPIO.LOW)
            GPIO.output(LEDpin_3,GPIO.LOW)
            GPIO.output(LEDpin_4,GPIO.LOW)
    if GPIO.input(BtnPin_3):
        time.sleep(0.04)
        while GPIO.input(BtnPin_3):
            GPIO.output(LEDpin_3,GPIO.HIGH)
    if GPIO.input(BtnPin_4):
        time.sleep(0.04)
        while GPIO.input(BtnPin_4):
            GPIO.output(LEDpin_4,GPIO.HIGH)
            GPIO.output(LEDpin_1,GPIO.LOW)
            GPIO.output(LEDpin_2,GPIO.LOW)
            GPIO.output(LEDpin_3,GPIO.LOW)
    GPIO.output(LEDpin_1,GPIO.LOW)
    GPIO.output(LEDpin_2,GPIO.LOW)
    GPIO.output(LEDpin_3,GPIO.LOW)
    GPIO.output(LEDpin_4,GPIO.LOW)
        
GPIO.cleanup()

File: Lesson/Lesson 24 Four-way answerer experiment/Java/Button.java
import com.pi4j.wiringpi.Gpio;

public class Button {
	static int BtnPin_1 = 1, LEDpin_1 =	4, BtnPin_2 = 24, LEDpin_2 = 5, BtnPin_3 = 28, LEDpin_3 = 6, BtnPin_4 =	29, LEDpin_4 =	25;
	
	static {
        if (Gpio.wiringPiSetup() == -1) {
            System.out.println(" ==>> GPIO SETUP FAILED");
        }
			
		Gpio.pinMode(BtnPin_1, Gpio.INPUT);               
		Gpio.pinMode(LEDpin_1, Gpio.OUTPUT);                  
		Gpio.pinMode(BtnPin_2, Gpio.INPUT);                   
		Gpio.pinMode(LEDpin_2, Gpio.OUTPUT);                  
		Gpio.pinMode(BtnPin_3, Gpio.INPUT);                   
		Gpio.pinMode(LEDpin_3, Gpio.OUTPUT);                  
		Gpio.pinMode(BtnPin_4, Gpio.INPUT);                   
		Gpio.pinMode(LEDpin_4, Gpio.OUTPUT);                  
		Gpio.digitalWrite(LEDpin_1, Gpio.HIGH);               
		Gpio.digitalWrite(LEDpin_2, Gpio.HIGH);               
		Gpio.digitalWrite(LEDpin_3, Gpio.HIGH);               
		Gpio.digitalWrite(LEDpin_4, Gpio.HIGH);		        
    }

	public static void main(String args[]) {
		while (true)
		{
			if(!(Gpio.digitalRead(BtnPin_1) == 1))
			{
				Gpio.delay(20);
				while(!(Gpio.digitalRead(BtnPin_1) == 1))
				{
					Gpio.digitalWrite(LEDpin_1,Gpio.LOW );
				}
				Gpio.digitalWrite(LEDpin_1,Gpio.HIGH );
			}
	
			if(!(Gpio.digitalRead(BtnPin_2) == 1))
			{
				Gpio.delay(20);
				while(!(Gpio.digitalRead(BtnPin_2) == 1))
				{
					Gpio.digitalWrite(LEDpin_2,Gpio.LOW );
				}
				Gpio.digitalWrite(LEDpin_2,Gpio.HIGH );
			}
			
			if(!(Gpio.digitalRead(BtnPin_3) == 1))
			{
				Gpio.delay(20);
				while(!(Gpio.digitalRead(BtnPin_3) == 1))
				{
					Gpio.digitalWrite(LEDpin_3,Gpio.LOW );
				}
				Gpio.digitalWrite(LEDpin_3,Gpio.HIGH );
			}
			
			if(!(Gpio.digitalRead(BtnPin_4) == 1))
			{
				Gpio.delay(20);
				while(!(Gpio.digitalRead(BtnPin_4) == 1))
				{
					Gpio.digitalWrite(LEDpin_4,Gpio.LOW );
				}
				Gpio.digitalWrite(LEDpin_4,Gpio.HIGH );
			}
		}
    }
}
